akka-http-core/src/main/scala/akka/http/impl/model/parser/SimpleHeaders.scala

```diff
@@ -95,7 +95,15 @@ private[parser] trait SimpleHeaders { this: Parser with CommonRules with CommonA
 
   // https://tools.ietf.org/html/rfc6265#section-4.2
   def `cookie` = rule {
-    oneOrMore(`cookie-pair`).separatedBy((ch(';') | ch(',')) ~ OWS) ~ EOI ~> (`Cookie`(_))
+    oneOrMore(`optional-cookie-pair`).separatedBy(';' ~ OWS) ~ EOI ~> { pairs ⇒
+      val validPairs = pairs.collect { case Some(p) ⇒ p }
+      `Cookie` {
+        if (validPairs.nonEmpty) validPairs
+        // Parsing infrastructure requires to return an HttpHeader value here but it is not possible
+        // to create a Cookie header without elements, so we create a dummy instance.
+        else HttpCookiePair("__dummy__", "__No_more_cookies_left_after_discarding_invalid_ones__") :: Nil
```

Comment:
> Actually it might even suffice to simply throw HeaderParser.RuleNotFoundException

No, this doesn't work and fails even worse because the error handling logic cannot deal with exceptions without messages (fails in `HeaderParser.failure` -> `ErrorInfo.fromCompoundString`).

I now throw a new dedicated exception which should do the trick.